test/test_main.py

```diff
@@ -0,0 +1,199 @@
+    'ftp_port': '4425',
+    'icone_ftp_username': 'username',
+    'icone_ftp_password': 'password',
+    'ftp_icone_file_path': 'test_filepath',    
+    })
+@patch.object(main, 'get_ftp_credentials')
+def test_get_ftp_url(ftp_credentials):
+    credentials='username', 'password'
+    main.get_ftp_credentials=MagicMock(return_value=credentials)
+    test_ftp_url='ftp://username:password@www.icone.com:4425/test_filepath'
+    actual=main.get_ftp_url()
+    assert actual==test_ftp_url
+
+@patch.object(main, 'get_ftp_credentials')
+def test_get_ftp_url_missing_environment_variable(ftp_credentials):
+    credentials='username', 'password'
+    main.get_ftp_credentials=MagicMock(return_value=credentials)
+    try:
+        actual=main.get_ftp_url()
+        assert False
+    except KeyError:
+        assert True
+
+#--------------------------------------------------------------------------------unit test for parse_xml function--------------------------------------------------------------------------------
+def test_parse_xml():
+    test_parse_xml_string= """<incident id="U13631595_202012160845">
+        <updatetime>2020-12-16T17:18:00Z</updatetime>
+        <type>CONSTRUCTION</type>
+         <type> Hazard </type>
+          <polyline>34.8380671,-114.1450650,34.8380671,-114.1450650</polyline>
+        </incident>"""
+    test_valid_output= {"incident": OrderedDict({'@id': 'U13631595_202012160845', 'updatetime': '2020-12-16T17:18:00Z', 'type': ['CONSTRUCTION', 'Hazard'], 'polyline': '34.8380671,-114.1450650,34.8380671,-114.1450650' })}
+    actual_output= main.parse_xml(test_parse_xml_string)
+    assert actual_output == test_valid_output
+
+#--------------------------------------------------------------------------------unit test for get_ftp_credentials function--------------------------------------------------------------------------------
+@patch.dict(os.environ, {
+    'icone_ftp_username_secret_name': 'secret_username',
+    'icone_ftp_password_secret_name': 'secret_password',
+    'project_id': 'project_id'})
+@patch('google.cloud.secretmanager.SecretManagerServiceClient')
+def test_get_ftp_credentials(secret):
+    secret().access_secret_version = fake_secret_client
+    actual = main.get_ftp_credentials()
+    valid_username = 'username'
+    valid_password = 'password'
+    expected = (valid_username, valid_password) 
+    assert actual == expected
+    
+@patch.dict(os.environ, {})
+@patch('google.cloud.secretmanager.SecretManagerServiceClient')
+def test_get_ftp_credentials_no_env_vars(secret):
+    secret().access_secret_version = fake_secret_client
+    actual = main.get_ftp_credentials()
+    assert actual == None
+
+@patch.dict(os.environ, {
+    'icone_ftp_username_secret_name': 'fail',
+    'icone_ftp_password_secret_name': 'secret_password',
+    'project_id': 'project_id'})
+@patch('google.cloud.secretmanager.SecretManagerServiceClient')
+def test_get_ftp_secrets_does_not_exist(secret):
```

Comment:
This tests the case that the "username" secret doesn't exist. Can we name it such to clarify? Also, since we have it handled separately, can we add a test for missing "password" secret?

Reply:
added a test case for password and renamed the other test case to test_get_ftp_secrets_username_does_not_exist